src/sage/graphs/graph_decompositions/slice_decomposition.pyx

```diff
@@ -0,0 +1,970 @@
+# cython: binding=True
+# distutils: language = c++
+# distutils: extra_compile_args = -std=c++11
+r"""
+Slice decomposition
+
+This module implements an extended lexBFS algorithm for computing the slice
+decomposition of undirected graphs and the class :class:`~SliceDecomposition` to
+represent such decompositions.
+
+AUTHORS:
+
+- Cyril Bouvier (2024-06-25): initial version
+"""
+# ****************************************************************************
+# Copyright (C) 2024 Cyril Bouvier <cyril.bouvier@lirmm.fr>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from libcpp.algorithm cimport swap
+from cython.operator cimport dereference as deref
+
+from sage.graphs.base.c_graph cimport CGraphBackend
+from sage.data_structures.bitset_base cimport bitset_in
+
+cdef void extended_lex_BFS(
+        CGraph cg, vector[int] &sigma, vector[int] *sigma_inv,
+        int initial_v_int, vector[int] *pred, vector[size_t] *xslice_len,
+        vector[vector[int]] *lex_label) except *:
+    r"""
+    Perform a extended lexicographic breadth first search (LexBFS) on the
```

Comment:
```diff
-    Perform a extended lexicographic breadth first search (LexBFS) on the
-    undirected graph `G`. In addition to computing a LexBFS ordering, the
-    extended LexBFS algorithm can be used to compute the slice decomposition of
-    the graph.
+    Perform an extended lexicographic breadth first search (LexBFS) on the
+    undirected graph ``cg``. 
+
+    In addition to computing a LexBFS ordering, the extended LexBFS algorithm
+    can be used to compute the slice decomposition of the graph.
```

Reply:
done in 201cbffcf4067690db6449d855214320e48fd2e7